checker/src/org/checkerframework/checker/index/IndexMethodIdentifier.java

```diff
@@ -65,18 +77,49 @@ public boolean isRandomNextInt(Tree tree, ProcessingEnvironment processingEnv) {
         return TreeUtils.isMethodInvocation(tree, fcnNextInt, processingEnv);
     }
 
-    public boolean isStringLength(Tree tree, ProcessingEnvironment processingEnv) {
-        return TreeUtils.isMethodInvocation(tree, stringLength, processingEnv);
+    /**
+     * @return whether or not {@code tree} is an invocation of a method that returns the length of
+     *     "this"
+     */
+    public boolean isLengthOfMethodInvocation(Tree tree) {
+        if (tree.getKind() != Kind.METHOD_INVOCATION) {
+            return false;
+        }
+        return isLengthOfMethodInvocation(TreeUtils.elementFromUse((MethodInvocationTree) tree));
+    }
+
+    /**
+     * @return whether or not {@code tree} is an invocation of a method that returns the length of
+     *     "this"
+     */
+    public boolean isLengthOfMethodInvocation(ExecutableElement ele) {
+        if (stringLength.equals(ele)) {
+            return true;
+        }
+
+        AnnotationMirror len = factory.getDeclAnnotation(ele, LengthOf.class);
+        if (len == null) {
+            return false;
+        }
+        List<String> values =
+                AnnotationUtils.getElementValueArray(len, "value", String.class, false);
+        if (values.size() == 1) {
```

Comment:
Is there a reason not to just loop over all of the values and check if any of them are `this`? I doubt this will come up much in practice, but it would be nice not to require `this` to be the only argument to the `@LengthOf` annotation.